src/main/java/uk/gov/ons/registers/cdap/service/CHService.java

```diff
@@ -0,0 +1,131 @@
+package uk.gov.ons.registers.cdap.service;
+
+import co.cask.cdap.api.annotation.UseDataSet;
+import co.cask.cdap.api.dataset.table.Get;
+import co.cask.cdap.api.dataset.table.Row;
+import co.cask.cdap.api.dataset.table.Scanner;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.api.service.AbstractService;
+import co.cask.cdap.api.service.http.AbstractHttpServiceHandler;
+import co.cask.cdap.api.service.http.HttpServiceRequest;
+import co.cask.cdap.api.service.http.HttpServiceResponder;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CHService extends AbstractService {
```

Comment:
Please see general comment on README.md about renaming CHService (and anything similar) from 'CH' to 'CompanyHouse' ('CompaniesHouse'?)